### ISSUE TITLE
Respect transparency dependencies

### ISSUE THREAD

**Issue by AlexV525** [google/json_serializable.dart#1025]
## Related issues
- #1011
- #1014
- #1020

## Summary
In 6.0.0, we dropped the transparent dependencies recognition for at least `json_annotation`, a package that should be defined in the `dependencies` section in the pubspec. This makes modular projects fail to use the annotation package from the base package.

Transparent dependencies are features for the pub, until the pub drops the support for this, please don't constraint this separately. It's been caused us to make our project annotate a large number of redundant dependencies.

**Comment by AlexV525** [comment]
@kevmoo

**Comment by kevmoo** [comment]
@AlexV525 – I'm not sure exactly what you want.

What do you want to do? What can't you do? Include code/yaml if you can.

**Comment by AlexV525** [comment]
Like the previous example: https://github.com/AlexV525/json_annotation_test.

TL;DR, we can use a base/upstream package's dependency, if it provides a valid version.

**Comment by AlexV525** [comment]
Thanks for the help. Now it works perfectly. 💙

**Comment by sgehrman** [comment]
Getting this.

[WARNING] json_serializable:json_serializable on lib/src/panels/home/daily_quote_model.dart:
The version constraint "^4.3.0" on json_annotation allows versions before 4.4.0 which is not allowed.

**Comment by Mugilan-Codes** [comment]
> Getting this.
> 
> [WARNING] json_serializable:json_serializable on lib/src/panels/home/daily_quote_model.dart: The version constraint "^4.3.0" on json_annotation allows versions before 4.4.0 which is not allowed.

Same I am getting this warning

`The version constraint "^4.6.0" on json_annotation allows versions before 4.8.0 which is not allowed.`

**Comment by egorozh** [comment]
I'm getting warning:
```You are missing a required dependency on json_annotation in the "dependencies" section of your pubspec with a lower bound of at least "4.9.0".```

base project pubspec:
```
dependencies:
  json_annotation: 4.9.0
dev_dependencies:
  json_serializable: 6.8.0
```
root project pubspec:
```
dependencies:
  root_proj:
    path: ../root_proj
dev_dependencies:
  json_serializable: 6.8.0
```
